PWGJE/TableProducer/jettaggerhf.cxx

```diff
@@ -21,64 +22,113 @@
+template <typename JetTableData, typename JetTableMCD, typename JetTaggingTableData, typename ConstTaggingTableData, typename ConstTCTaggingTableData, typename JetTaggingTableMCD, typename ConstTaggingTableMCD, typename ConstTCTaggingTableMCD>
 struct JetTaggerHFTask {
 
   Produces<JetTaggingTableData> taggingTableData;
+  Produces<ConstTaggingTableData> constTableData;
+  Produces<ConstTCTaggingTableData> constTCTableData;
   Produces<JetTaggingTableMCD> taggingTableMCD;
+  Produces<ConstTaggingTableMCD> constTableMCD;
+  Produces<ConstTCTaggingTableMCD> constTCTableMCD;
 
+  Configurable<bool> doWShower{"doWShower", false, "find jet origin included gluon spliting"}; // true:: remove gluon spliting
   Configurable<bool> doAlgorithm1{"doAlgorithm1", false, "fill table for algoithm 1"};
   Configurable<bool> doAlgorithm2{"doAlgorithm2", false, "fill table for algoithm 2"};
   Configurable<bool> doAlgorithm3{"doAlgorithm3", false, "fill table for algoithm 3"};
   Configurable<float> maxDeltaR{"maxDeltaR", 0.25, "maximum distance of jet axis from flavour initiating parton"};
 
+  using JetTagTracksData = soa::Join<aod::JTracks, aod::JTrackTagDcas, aod::JTrackTagDcaCovs>;
+  using JetTagTracksMC = soa::Join<aod::JTracks, aod::JMcTrackLbs, aod::JTrackTagDcas, aod::JTrackTagDcaCovs>;
+
   void processDummy(aod::Collision const& collision)
   {
   }
   PROCESS_SWITCH(JetTaggerHFTask, processDummy, "Dummy process", true);
 
-  void processData(aod::JCollision const& collision, JetTableData const& jets, aod::JTracks const& tracks)
+  void processData(soa::Join<aod::JCollisions, aod::JCollisionPIs>::iterator const& jcollision, aod::Collisions&, JetTableData const& jets, aod::JTracks&, JetTagTracksData const& tracks)
   {
+    auto collision = jcollision.template collision_as<aod::Collisions>();
     for (auto& jet : jets) {
-
       int algorithm1 = jet.globalIndex(); // This needs to be changed. It is only done because O2Physics compilation breaks if jet is unused
       int algorithm2 = 0;
       int algorithm3 = 0;
       // if (doAlgorithm1) algorithm1 = JetTaggingUtilities::Algorithm1((mcdjet, tracks);
       // if (doAlgorithm2) algorithm2 = JetTaggingUtilities::Algorithm2((mcdjet, tracks);
       // if (doAlgorithm3) algorithm3 = JetTaggingUtilities::Algorithm3((mcdjet, tracks);
       taggingTableData(0, algorithm1, algorithm2, algorithm3);
+      std::vector<int> trackconst;
+      std::vector<int> vecGeoSign;
+      std::vector<float> vecTrackPt, vecTrackEta, vecTrackPhi, vecSignedIP2D, vecSignedIP2Ds, vecSignedIP3D, vecSignedIP3Ds;
+      for (auto& track : jet.template tracks_as<JetTagTracksData>()) {
+        trackconst.push_back(track.globalIndex());
+        int geoSign = JetTaggingUtilities::getGeoSign(collision, jet, track);
+        trackconst.push_back(track.globalIndex());
+        vecGeoSign.push_back(geoSign);
+        vecTrackPt.push_back(track.pt());
+        vecTrackEta.push_back(track.eta());
```

Comment:
this is fine if you intend to store these tables later for offline processing. If not you can just put an index to the track table as you have above aand then you dont need to save pt, eta and phi of tracks